roles/converter-filters/tasks/main.yml

```diff
@@ -0,0 +1,158 @@
+- name: check existing stack
+  stat:
+    path: /usr/local/bin/stack
+  register: stack
+
+- name: Download stack
+  get_url:
+    url: https://github.com/commercialhaskell/stack/releases/download/v1.7.1/stack-1.7.1-linux-x86_64.tar.gz
+    dest: /tmp/stack-1.7.1-linux-x86_64.tar.gz
+    mode: 0660
+  when: converter == 1 and not stack.stat.exists and converter_build == 1
+  tags:
+     - setup
+
+- name: extract stack
+  unarchive:
+    src: /tmp/stack-1.7.1-linux-x86_64.tar.gz
+    dest: /tmp
+    remote_src: yes
+    exclude: doc,LICENSE
+  when: converter == 1 and not stack.stat.exists and converter_build == 1
+  tags:
+     - setup
+
+- name: install stack
+  copy:
+    src: /tmp/stack-1.7.1-linux-x86_64/stack
+    dest: /usr/local/bin
+    mode: 0755
+    remote_src: yes
+  when: converter == 1 and not stack.stat.exists and converter_build == 1
+  tags:
+     - setup
+
+- name: support packages
+  apt: pkg={{ item }} state=present update_cache=true install_recommends=no
+  with_items:
+    - libgmp3-dev
+  when: app == 1
+  tags:
+     - setup
+
+- name: setup checkout path
+  file: path="{{ converter_checkout_path }}" owner="{{ converter_user }}" state=directory
+  when: converter == 1 and converter_build == 1
```

Comment:
consider blocks ;) @yakky https://www.google.it/search?q=ansible+blocks&oq=ansible+blocks&aqs=chrome..69i57j69i60j0l4.2392j0j7&sourceid=chrome&ie=UTF-8